src/primitives/poseidon.rs

```diff
@@ -6,7 +6,7 @@ use std::fmt;
 use std::iter;
 use std::marker::PhantomData;
 
-use halo2::arithmetic::FieldExt;
+use halo2_proofs::arithmetic::FieldExt;
```

Comment:
`src/primitives/poseidon*` should be deleted; it is provided by `halo2_gadgets` now (since we _need_ the implementation over there).